File: angular/src/app/data-search/entities/statistics-fields.ts
export class StatisticsFields {
  // selectable statistics types
  private statisticsTypes: string[] = null;
  // selectable group by fields
  private groupByFields: string[] = null;
  // selectable sub group by fields
  private groupBySubFields: string[] = null;
  // selectable compute fields
  private computeFields: string[] = null;
  // getter
  public getStatisticsTypes(): string[] {
    return this.statisticsTypes;
  }
  // setter
  public setStatisticsTypes(statisticsTypes: string[]) {
    this.statisticsTypes = statisticsTypes;
  }
  // getter
  public getGroupByFields(): string[] {
    return this.groupByFields;
  }
  public setGroupByFields(groupByFields: string[]) {
    this.groupByFields = groupByFields;
  }
  // getter
  public getGroupBySubFields(): string[] {
    return this.groupBySubFields;
  }
  public setGroupBySubFields(groupBySubFields: string[]) {
    this.groupBySubFields = groupBySubFields;
  }
  // getter
  public getComputeFields(): string[] {
    return this.computeFields;
  }
  // setter
  public setComputeFields(computeFields: string[]) {
    this.computeFields = computeFields;
  }
}

File: spring/src/main/java/com/example/eurasia/entity/Data/StatisticsFields.java
package com.example.eurasia.entity.Data;

public class StatisticsFields {
    // pie chart, line chart, bar chart
    private String[] statisticsTypes = null;
    // group by field
    private String[] groupByFields = null;
    // group by sub field
    private String[] groupBySubFields = null;
    // compute fields
    private String[] computeFields = null;
    // set statistics types

    public void setStatisticsTypes(String[] statisticsTypes) {
        this.statisticsTypes = statisticsTypes;
    }
    public String[] getStatisticsTypes(){
        return this.statisticsTypes;
    }
    public void setGroupByFields(String[] groupByFields){
        this.groupByFields = groupByFields;
    }
    public String[] getGroupByFields(){
        return this.groupByFields;
    }
    public void setGroupBySubFields(String[] groupBySubFields){
        this.groupBySubFields = groupBySubFields;
    }
    public String[] getGroupBySubFields(){
        return this.groupBySubFields;
    }
    public void setComputeFields(String[] computeFields){
        this.computeFields = computeFields;
    }
    public String[] getComputeFields(){
        return computeFields;
    }
}
